addons/account/models/account_move.py

```diff
@@ -4058,6 +4058,8 @@ def create(self, vals_list):
         BUSINESS_FIELDS = ('price_unit', 'quantity', 'discount', 'tax_ids')
 
         for vals in vals_list:
+            if vals.get('display_type', False):
+                vals.update(quantity=0)
```

Comment:
```suggestion
            if vals.get('display_type'):
                vals['quantity'] = 0

Reply:
I would prefer if it was done in `default_get` or in a lambda function on the field though (depending on the context key `default_display_type`) so that it is more explicit regarding the purpose of this and why it is needed.